app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
```

Comment:
I like @sbc100 suggestion of splitting it into 2 different methods, for assignment and access. That's much clearer to me.